core/market/decentralized/migrations/2020-05-15-093513_init-db/up.sql

```diff
@@ -26,6 +26,85 @@ CREATE TABLE market_demand (
 	node_id VARCHAR(20) NOT NULL,
 
 	creation_ts DATETIME NOT NULL,
-	insertion_ts DATETIME NOT NULL DEFAULT CURRENT_TIMESTAMP,
+	insertion_ts DATETIME NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
 	expiration_ts DATETIME NOT NULL
 );
+
+CREATE TABLE market_event_type(
+    id INTEGER NOT NULL PRIMARY KEY,
+    event_type VARCHAR(50) NOT NULL,
+    role VARCHAR(10) NOT NULL CHECK (role in ('Requestor', 'Provider'))
+);
+
+INSERT INTO market_event_type(id, event_type, role) VALUES
+    (1001, "Proposal", "Provider"),
+    (1002, "Agreement", "Provider"),
+    (1003, "PropertyQuery", "Provider"),
+    (2001, "Proposal", "Requestor"),
+    (2002, "PropertyQuery", "Requestor");
+
+CREATE TABLE market_requestor_event(
+    id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+    subscription_id INTEGER NOT NULL,
+    timestamp DATETIME NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
+    event_type INTEGER NOT NULL,
+    artifact_id VARCHAR(100) NOT NULL,
+
+    FOREIGN KEY(event_type) REFERENCES market_event_type (id),
+    CHECK (event_type>=2000)
+);
+
+CREATE TABLE market_provider_event(
+    id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+    subscription_id INTEGER NOT NULL,
+    timestamp DATETIME NOT NULL DEFAULT(STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW')),
+    event_type INTEGER NOT NULL,
+    artifact_id VARCHAR(100) NOT NULL,
+
+    FOREIGN KEY(event_type) REFERENCES market_event_type (id),
+    CHECK (event_type<2000)
+);
+
+CREATE TABLE market_proposal_state(
+    id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+    state VARCHAR(10) NOT NULL UNIQUE
+);
+
+INSERT INTO market_proposal_state(id, state) VALUES
+    (0, "Initial"),
+    (1, "Draft"),
+    (2, "Rejected"),
+    (3, "Accepted"),
+    (4, "Expired");
+
+CREATE TABLE market_negotiation(
+    id VARCHAR(100) NOT NULL PRIMARY KEY,
+    subscription_id VARCHAR(97) NOT NULL,
+
+    offer_id VARCHAR(97) NOT NULL,
+    demand_id VARCHAR(97) NOT NULL,
+
+    identity_id VARCHAR(20) NOT NULL,
```

Comment:
what is the relation of `identity_id` and below ids?

Reply:
Check structures in db::model first. Everything is described there in details. I didn't left comments in sql files